### PR TITLE
As an API user, I can change the status of a Finding

### PR DESCRIPTION
**Pull request by filewalkwithme**
This PR introduces two new endpoints in the Vulcan API, allowing the user to submit Finding overwrites.
A finding overwrite is an event that changes attributes of a finding.
For the time being, the user will be allowed to change the `STATUS` field only. Also, the user will be required to provide additional `notes`, explaining why the overwrite is happening.

```
LIST OF NEW ENDPOINTS

# Submit a Finding overwrite event
POST /api/v1/teams/{team_id}/findings/{finding_id}/overwrites

# Get a list of Finding overwrite events
GET  /api/v1/teams/{team_id}/findings/{finding_id}/overwrites
```

### Submit a Finding overwrite event
When receiving a new Finding Overwrite, the Vulcan API will start a distributed transaction:
- It will keep a record of the event in it own's database and,
- Create a new record in the CDC table. Later on, a Go routine will parse this record and send a request to the Vulnerability DB.

### Get a list of Finding overwrite events
This endpoint will retrieve the content of the FINDING_OVERWRITE table for a specific finding.

### CODE REVIEW

**Review by ka3de — CHANGES_REQUESTED** [review]
Hey @filewalkwithme , thanks for working on this!

Overall I found it pretty clear, just some considerations.
Also, I think the metrics middleware actions and entities mapping is missing for the new endpoints.
And, this might be a pain but..., as a general consideration for the PR, should we use the word `override` or `overwrite` ? [ref](https://www.grammar.com/override_vs_overwrite).

Another question that I have is, what are the plans for this new `override` endpoint and DB table? The plan is to extend it also to store other "status" changes for the findings like "snooze" ?

File: pkg/api/service/finding.go [hunk 1]
```diff
@@ -33,3 +36,47 @@ func (s vulcanitoService) ListFindingsByTarget(ctx context.Context, params api.F
...
+	// FALSE_POSITIVE -> OPEN
+	// OPEN           -> FALSE_POSITIVE
+	// FALSE_POSITIVE -> FALSE_POSITIVE
+	if findingOverride.StatusPrevious != "OPEN" && findingOverride.StatusPrevious != "FALSE_POSITIVE" {
+		return errors.Validation(fmt.Sprintf("Status transition not allowed: from '%s' to '%s'", findingOverride.StatusPrevious, findingOverride.Status))
+	}
+
+	if findingOverride.Status != "OPEN" && findingOverride.Status != "FALSE_POSITIVE" {
+		return errors.Validation(fmt.Sprintf("Status transition not allowed: from '%s' to '%s'", findingOverride.StatusPrevious, findingOverride.Status))
+	}
```
[the remark this answers is not in the document]
Comment (filewalkwithme) [reply pkg/api/service/finding.go:62]: Done!

File: pkg/api/transport/routes.go [hunk 2]
```diff
@@ -130,6 +130,8 @@ func AttachRoutes(e endpoint.Endpoints, logger kitlog.Logger) http.Handler {
 	r.Methods("GET").Path("/api/v1/teams/{team_id}/findings/targets").Handler(newServer(e[endpoint.ListFindingsTargets], endpoint.FindingsRequest{}, logger, endpoint.ListFindingsTargets))
 	r.Methods("GET").Path("/api/v1/teams/{team_id}/findings/targets/{target_id}").Handler(newServer(e[endpoint.ListFindingsByTarget], endpoint.FindingsByTargetRequest{}, logger, endpoint.ListFindingsByTarget))
 	r.Methods("GET").Path("/api/v1/teams/{team_id}/findings/{finding_id}").Handler(newServer(e[endpoint.FindFinding], endpoint.FindingsRequest{}, logger, endpoint.FindFinding))
+	r.Methods("GET").Path("/api/v1/teams/{team_id}/findings/{finding_id}/override").Handler(newServer(e[endpoint.ListFindingOverrides], endpoint.FindingsRequest{}, logger, endpoint.ListFindingOverrides))
+	r.Methods("POST").Path("/api/v1/teams/{team_id}/findings/{finding_id}/override").Handler(newServer(e[endpoint.CreateFindingOverride], endpoint.FindingOverrideRequest{}, logger, endpoint.CreateFindingOverride))
```
[the remark this answers is not in the document]
Comment (filewalkwithme) [reply pkg/api/transport/routes.go]: Done!

File: pkg/api/store/cdc/parser.go [hunk 3]
```diff
@@ -72,6 +73,8 @@ func (p *VulnDBTxParser) Parse(log []Event) (nParsed uint) {
 			processFunc = p.processDeleteAsset
 		case opDeleteAllAssets:
 			processFunc = p.processDeleteAllAssets
+		case opFindingOverride:
+			processFunc = p.findingOverride
```
[the remark this answers is not in the document]
Comment (filewalkwithme) [reply pkg/api/store/cdc/parser.go]: Done!

File: pkg/api/endpoint/findings.go [hunk 4]
```diff
@@ -226,6 +226,80 @@ func makeFindFindingEndpoint(s api.VulcanitoService, logger kitlog.Logger) endpo
...
+		if authorizedFindFindingRequest(finding.Finding.Target.Tags, team.Tag) {
+			err := s.CreateFindingOverride(ctx, findingOverride)
+			if err != nil {
+				return nil, err
+			}
+
+			return Ok{}, nil
+		}
+
+		return Forbidden{nil}, nil
```
[the remark this answers is not in the document]
Comment (filewalkwithme) [reply pkg/api/endpoint/findings.go]: Done!

File: db/sql/V1.49__create_findings_overrides.sql [hunk 5]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE finding_overrides (
+    id              UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    user_id         UUID NOT NULL,
+    finding_id      UUID NOT NULL,
```
[the remark this answers is not in the document]
Comment (filewalkwithme) [reply db/sql/V1.49__create_findings_overrides.sql]: I have added the index.

**Comment by filewalkwithme** [comment]
@ka3de 

> Also, I think the metrics middleware actions and entities mapping is missing for the new endpoints.

I have mapped the new FindingOverwrite endpoints to the existing `entityFinding` entity. Let me know if think that makes more sense to create a specific entity for the `FindingOverwrite`.

> And, this might be a pain but..., as a general consideration for the PR, should we use the word `override` or `overwrite` ? [ref](https://www.grammar.com/override_vs_overwrite).

Not strong opinion on this. I just modified to PR to use the term `overwrite` as you asked.

> Another question that I have is, what are the plans for this new `override` endpoint and DB table? The plan is to extend it also to store other "status" changes for the findings like "snooze" ?

I do believe that the "snooze" feature will be implemented using some sort of `tagging` system in the Vulnerability DB. Of course, we will be required to log this action in somewhere and for the moment we have decide to track this at Vulcan API level. However, for me it's not clear if we are going to use the same table. Let's discuss this in a future discussion.

**Review by ka3de — APPROVED** [review]
LGTM.

**Pull request merged**